samcli/local/services/base_local_service.py

```diff
@@ -118,4 +121,19 @@ def get_lambda_output(stdout_stream):
             # Last line is Lambda response. Make sure to strip() so we get rid of extra whitespaces & newlines around
             lambda_response = stdout_data[last_line_position:].strip()
 
-        return lambda_response, lambda_logs
+        is_lambda_user_error_response = False
+
+        try:
+            lambda_response_dict = json.loads(lambda_response)
+
+            if isinstance(lambda_response_dict, dict) and \
+                    len(lambda_response_dict.keys()) == 3 and \
+                    'errorMessage' in lambda_response_dict and \
+                    'errorType' in lambda_response_dict and \
```

Comment:
`errorType` and `stackTrace` are required. For example, this just returns `{"errorMessage": "error" }` (I verified by calling Lambda). But if you returned a regular exception like `TypeError`, then the other two fields are returned.

```javascript
exports.handler = (event, context, callback) => {
    // TODO implement
    callback('error');
};
```